### PR TITLE
Add new API version 2020-09-01

### PR DESCRIPTION
**Pull request by xingwu1**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json [hunk 1]
```diff
@@ -5951,7 +5951,7 @@
             },
             "description": "A response containing the Task counts for the specified Job.",
             "schema": {
-              "$ref": "#/definitions/TaskCounts"
+              "$ref": "#/definitions/TaskCountsResult"
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json:5954]: is this intentional?

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json [hunk 2]
```diff
@@ -11798,6 +11801,12 @@
           "$ref": "#/definitions/TaskConstraints",
           "title": "Constraints that apply to the Job Manager Task."
         },
+        "requiredSlots": {
+          "type": "integer",
+          "format": "int32",
+          "title": "The number of scheduling slots that the Task required to run.",
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json:11807]:
```suggestion
          "title": "The number of scheduling slots that the Task requires to run.",
```

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json [hunk 3]
```diff
@@ -13328,11 +13391,11 @@
...
+          "description": "This property is mutually exclusive with other ImageReference properties. The Shared Image Gallery Image must have replicas in the same region and must be in the same subscription as the Azure Batch account. If the image version is not specified in the imageId, the latest version will be used. For information about the firewall settings for the Batch Compute Node agent to communicate with the Batch service see https://docs.microsoft.com/en-us/azure/batch/batch-api-basics#virtual-network-vnet-and-firewall-configuration."
         }
       },
-      "title": "A reference to an Azure Virtual Machines Marketplace Image or a custom Azure Virtual Machine Image. To get the list of all Azure Marketplace Image references verified by Azure Batch, see the 'List supported Images' operation."
+      "title": "A reference to an Azure Virtual Machines Marketplace Image or a bring-your-own Shared Image Gallery Image. To get the list of all Azure Marketplace Image references verified by Azure Batch, see the 'List Supported Images' operation."
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json:13398]: probably don't need bring-your-own here as its implied

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json [hunk 4]
```diff
@@ -14445,6 +14508,12 @@
           "$ref": "#/definitions/TaskConstraints",
           "title": "The execution constraints that apply to this Task."
         },
+        "requiredSlots": {
+          "type": "integer",
+          "format": "int32",
+          "title": "The number of scheduling slots that the Task required to run.",
```
Comment (bgklein) [inline specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json:14514]: requires here again

**Review by chiragg4u — APPROVED** [review]
Reviewed the swagger updates in the new version, they are looking ok. Please resolve the other comments and linter failures if any. Marking it ARMApproved.

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json (hunk 4 above)
Comment (xingwu1) [reply specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json]: fixed.

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json (hunk 3 above)
Comment (xingwu1) [reply specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json]: fixed.

File: specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json (hunk 2 above)
Comment (xingwu1) [reply specification/batch/data-plane/Microsoft.Batch/stable/2020-09-01.12.0/BatchService.json]: fixed.

**Comment by xingwu1** [comment]
@chiragg4u, I had updated the comments base on feedback. Please merge it when you are available.

**Comment by jhendrixMSFT** [comment]
@xingwu1 are these APIs live?

**Review by bgklein — APPROVED** [review]

**Comment by xingwu1** [comment]
@jhendrixMSFT , it is live on most regions. It is OK to let it merge, since we still need time to release SDK.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**